src/App.tsx

```diff
@@ -15,49 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = string;
+type Reverse = boolean;
+
+enum Sortfield {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+type Props = {
+  goods: Goods[];
+};
+
+const GoodsList: React.FC<Props> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
+function getPrepareGoods(
+  goods: Goods[],
+  sortGood: Sortfield | undefined,
+  reverse: Reverse,
+): Goods[] {
+  const prepareGoods: Goods[] = [...goods];
+
+  if (prepareGoods) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortGood) {
+        case Sortfield.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case Sortfield.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortGood, setSortGood] = useState<Sortfield>(Sortfield.DEFAULT);
+  const [reverse, setReverse] = useState<Reverse>(false);
+  const visibleGoods = getPrepareGoods(goodsFromServer, sortGood, reverse);
```

Comment:
make sense to use useMemo here, because it's essential to performance

Reply:
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/55200530/d4912b59-ff17-4a12-aa5e-eacbbc0c7606)

still not resolved